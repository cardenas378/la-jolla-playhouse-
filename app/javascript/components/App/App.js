import PropTypes from "prop-types";
import React from "react";
import {
  BrowserRouter as Router,
  Redirect,
  Route,
  Link
} from "react-router-dom";

import styled, { ThemeProvider } from "styled-components/macro";

import { Home } from "../../Pages/Home";
import { Locations } from "../../Pages/Locations";
import { AddBooking } from "../AddBooking";
import { AddLocation } from "../AddLocation";
import { getLocations, createLocation } from "../api";
import { NavBar } from "../NavBar";
import GlobalStyle from "../styles/Global";
import Theme from "../styles/Theme";

import { StyledLayout } from "./App.styles";

import {
  Header,
  AddBookingHeader,
  FormSectionHeader,
  StyledSelect,
  FullInput,
  RadioButton,
  SubmitButton,
  SaveBookingButton,
  HalfInput,
  ThirdInput
} from "..";

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      locations: [],
      success: false
    };
  }

  componentDidMount() {
    getLocations().then(locations => {
      this.setState({ locations });
    });
  }

  handleNewLocation = location => {
    createLocation(location).then(location => {
      this.setState({ success: true });
    });
  };

  changeSuccess = value => {
    this.setState({ success: value });
  };

  componentDidMount() {
    getLocations().then(locations => {
      this.setState({ locations });
    });
  }

  render() {
    const { locations, success } = this.state;

    return (
      <ThemeProvider theme={Theme}>
        <GlobalStyle />
        <StyledLayout>
          <Router>
            <NavBar />

            <main>
              <Route
                exact
                path="/AddLocation"
                render={props => {
                  return (
                    <AddLocation
                      {...props}
                      handleNewLocation={this.handleNewLocation}
                      success={success}
                    />
                  );
                }}
              />

              <Route
                exact
                path="/Locations"
                render={props => {
                  return (
                    <Locations
                      {...props}
                      changeSuccess={this.changeSuccess}
                      locations={locations}
                      success={success}
                    />
                  );
                }}
              />

              <Route
                exact
                path="/"
                render={props => {
                  return <Home {...props} />;
                }}
              />
              <Route
                exact
                path="/AddBooking/:location_id"
                render={props => {
                  return (
                    <AddBooking
                      {...props}
                      changeSuccess={this.changeSuccess}
                      locations={this.state.locations}
                      success={this.state.success}
                    />
                  );
                }}
              />
            </main>
          </Router>
        </StyledLayout>
      </ThemeProvider>
    );
  }
}

App.propTypes = {};

export { App };
