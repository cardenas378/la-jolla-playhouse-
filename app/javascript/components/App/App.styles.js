import styled from "styled-components";

const StyledLayout = styled.div`
  display: grid;
  grid-template-columns: 225px 1fr;
  grid-template-areas: "sidebar main";

  main {
    grid-area: main;
    background: ${props => props.theme.whiteSmoke};
    min-height: 100vh;
  }
`;

export { StyledLayout };
