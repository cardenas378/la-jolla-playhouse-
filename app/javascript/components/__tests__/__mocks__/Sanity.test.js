import Enzyme, { mount, shallow, render } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";
import ReactDOM from "react-dom";

import App from "../../App";

Enzyme.configure({ adapter: new Adapter() });
jest.mock("../../api/");

it("renders without crashing", () => {
  const div = document.createElement("div");

  ReactDOM.render(<App />, div);
});

describe("App", () => {
  test("renders the component correctly", () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});
