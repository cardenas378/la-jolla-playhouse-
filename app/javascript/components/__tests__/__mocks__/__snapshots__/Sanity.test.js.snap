// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the component correctly 1`] = `
<ThemeProvider
  theme={
    Object {
      "alabaster": "#fef217",
      "black": "#212121",
      "broom": "#fef217",
      "charcoal": "#424242",
      "crayonOrange": "#f26726",
      "doveGrey": "#757575",
      "error": "#b92121",
      "frenchGray": "#bdbdbd",
      "funBlue": "#1862a6",
      "georgia": "georgia, serif",
      "melon": "#fbbfb7",
      "nunito": "Nunito, sansSerif",
      "persianIndigo": "#66239c",
      "sansSerif": "Nunito, sansSerif",
      "serif": "georgia, serif",
      "starDust": "#9e9e9e",
      "stormDust": "#616161",
      "success": "#29a028",
      "warning": "#f7df5a",
      "weightBlack": 900,
      "weightBold": 700,
      "weightRegular": 400,
      "white": "#eeeeee",
      "whiteSmoke": "#f5f5f5",
      "zircon": "#e0e0e0",
    }
  }
>
  <GlobalStyleComponent />
  <app__StyledLayout>
    <BrowserRouter>
      <aside>
        <Logo
          className=""
          height="120"
          width="113"
        />
        <ol
          className="top-nav"
        >
          <li>
            <Link
              to="/"
            >
              Home
            </Link>
          </li>
          <li>
            <Link
              to="/AddLocation"
            >
              Add Location
            </Link>
          </li>
        </ol>
        <ol
          className="btm-nav"
        >
          <li>
            <Link>
              Settings
            </Link>
          </li>
          <li>
            <Link>
              Logout
            </Link>
          </li>
        </ol>
      </aside>
      <main>
        <Route
          exact={true}
          path="/AddLocation"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/"
          render={[Function]}
        />
      </main>
    </BrowserRouter>
  </app__StyledLayout>
</ThemeProvider>
`;
